### PR TITLE
joinPath() function issue #6012

### PR DESCRIPTION
**Pull request by victor-ludorum**
Adding **joinPath()** function to **path.chpl** file . This function is helpful in joining path names . 
@lydia-duncan Kindly review this , references #6012 and suggest necessary changes

### CODE REVIEW

**Comment by lydia-duncan** [comment]
Our Travis testing failed due to the presence of tabs

**Comment by victor-ludorum** [comment]
Okay @lydia-duncan I don't know about this . I will try to correct it .

**Comment by lydia-duncan** [comment]
It looks like you missed some in the update.

As a recommendation for code in general, try for this style:
```chapel
proc foo() {
  var x = something;
  for i in 1..10 {
    writeln(i); // contents of loops and if statements should be indented further for readability
  }
}
```

**Comment by victor-ludorum** [comment]
@lydia-duncan Okay I will follow that style .

**Comment by victor-ludorum** [comment]
@lydia-duncan  I have added **testfile** too in this error . Kindly review it and suggest the changes

File: test/library/standard/Path/victor-ludorum/joinPath/testcases.chpl [hunk 1]
```diff
@@ -0,0 +1,30 @@
+use Path;
+
+// writing tests for path joining where merging path name starts with ('/')
+writeln(joinPaths("/foo/bar", "/baz"));
```
Comment (lydia-duncan) [inline test/library/standard/Path/victor-ludorum/joinPath/testcases.chpl]: It looks like you missed updating the name here.  Did you try running this through start_test to ensure it worked?

Comment (victor-ludorum) [reply test/library/standard/Path/victor-ludorum/joinPath/testcases.chpl]: Oh sorry Mam I will update it .

File: test/library/standard/Path/victor-ludorum/joinPath/testcases.good [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/baz
+/foo/./baz
+/foo/./baz
+/foo/baz
+/baz
+/foo/./baz
+/foo/baz
+/foo
```
Comment (lydia-duncan) [inline test/library/standard/Path/victor-ludorum/joinPath/testcases.good]: I think this line should instead be "/foo/".  Did you run this test through start_test to check?

File: modules/standard/Path.chpl [hunk 3]
```diff
@@ -225,5 +225,40 @@ proc file.realPath(): string throws {
...
+    if temp.startsWith('/') {
+    result = temp;
+    }  
+    else if result.endsWith('/') {
+    result = result + temp;
+    }
+    else {
+     result = result + "/" + temp;
+       }
+     }
```
Comment (lydia-duncan) [inline modules/standard/Path.chpl]: The indentation for the for loop is a bit wonky.  Contents of a for loop or if statement should be slightly more indented than the loop's indentation, for clarity.
```chapel
if a {
  for i in 1..10 {
    var x = i;
    if (b) {
      var y = 10;
    }
  }
}
```

Comment (victor-ludorum) [reply modules/standard/Path.chpl]: Okay Mam I will correct it .

File: test/library/standard/Path/victor-ludorum/joinPath/testcases.good (hunk 2 above)
Comment (victor-ludorum) [reply test/library/standard/Path/victor-ludorum/joinPath/testcases.good]: Yeah Mam I have mistakenly added test for joinPath("/foo/") . I will correct it by adding test for 
joinPath("/foo")  not for joinPath("/foo/") .

**Comment by victor-ludorum** [comment]
@lydia-duncan I have updated the indentation of for loop . suggest if some more changes is needed

File: modules/standard/Path.chpl [hunk 4]
```diff
@@ -225,5 +225,40 @@ proc file.realPath(): string throws {
...
+            it, or all the paths provided if no absolute path is present
+   :rtype: `string`
+*/
+  proc joinPath(paths: string ...?n): string {
+    var result : string = paths(1); // result variable stores final answer
+   // loop to iterate over all the paths
+    for i in 2..n {
+      var temp : string = paths(i); 
+      if temp.startsWith('/') {
+      result = temp;
```
Comment (lydia-duncan) [inline modules/standard/Path.chpl]: All levels please, @victor-ludorum :)
```chapel
if (a) {
  result = temp;
}
```

Comment (victor-ludorum) [reply modules/standard/Path.chpl]: Okay mam I will .

**Comment by lydia-duncan** [comment]
Running a full standard paratest now.  I verified that the provided test case works as expected, and I am satisfied with its coverage.

In addition, this will not be merged until a contributor's agreement has been provided.  Thanks!

**Comment by victor-ludorum** [comment]
Thanks @lydia-duncan I will submit the contributor's agreement as soon as possible .

**Comment by lydia-duncan** [comment]
Full testing passed

**Comment by lydia-duncan** [comment]
Contributor agreement received.  Re-ran Travis testing after the license updates were merged into master and that passed as well (as expected).  Merging

**Pull request merged**